### PR TITLE
Implements Antag Player Client Age Restriction

### PR DESCRIPTION
**Pull request by Geevies**
Adds limits to how old a client should be on Aurora before they can possibly roll for an antag. Variable for each antagonist, based on difficulty and importance to round.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by sheeplets** [comment]
But how will I get parapenned and disposaled by /tg/ refugees now?

**Comment by XanderDox** [comment]
Don't know if this has happened already behind the scenes, but due to the fact this is about regulating gameplay conduct in an OOC sense, the Headmins @Alberyk and Garn should probably review the numbers in this.

**Comment by SonicGotNuked** [comment]
I hope this will help let new players experience the server before rolling something like malf. I think traitor would be fine as a starting antag.

File: code/game/antagonist/mutiny/mutineer.dm [hunk 1]
```diff
@@ -7,6 +7,7 @@ var/datum/antagonist/mutineer/mutineers
 	id = MODE_MUTINEER
 	antag_indicator = "mutineer"
 	restricted_jobs = list("Captain")
+	required_age = 3
```
Comment (Geevies) [inline code/game/antagonist/mutiny/mutineer.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/outsider/mercenary.dm [hunk 2]
```diff
@@ -12,6 +12,7 @@ var/datum/antagonist/mercenary/mercs
 	flags = ANTAG_OVERRIDE_JOB | ANTAG_CLEAR_EQUIPMENT | ANTAG_CHOOSE_NAME | ANTAG_HAS_NUKE | ANTAG_SET_APPEARANCE | ANTAG_HAS_LEADER
 	id_type = /obj/item/weapon/card/id/syndicate
 	antaghud_indicator = "hudoperative"
+	required_age = 7
```
Comment (Geevies) [inline code/game/antagonist/outsider/mercenary.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/outsider/ninja.dm [hunk 3]
```diff
@@ -10,6 +10,7 @@ var/datum/antagonist/ninja/ninjas
 	restricted_species = list("Diona")
 	flags = ANTAG_OVERRIDE_JOB | ANTAG_CLEAR_EQUIPMENT | ANTAG_CHOOSE_NAME | ANTAG_RANDSPAWN | ANTAG_VOTABLE | ANTAG_SET_APPEARANCE
 	antaghud_indicator = "hudninja"
+	required_age = 7
```
Comment (Geevies) [inline code/game/antagonist/outsider/ninja.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/outsider/raider.dm [hunk 4]
```diff
@@ -10,6 +10,7 @@ var/datum/antagonist/raider/raiders
 	welcome_text = "Use :H to talk on your encrypted channel."
 	flags = ANTAG_OVERRIDE_JOB | ANTAG_CLEAR_EQUIPMENT | ANTAG_CHOOSE_NAME | ANTAG_VOTABLE | ANTAG_SET_APPEARANCE | ANTAG_HAS_LEADER
 	antaghud_indicator = "hudmutineer"
+	required_age = 5
```
Comment (Geevies) [inline code/game/antagonist/outsider/raider.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/outsider/wizard.dm [hunk 5]
```diff
@@ -9,6 +9,7 @@ var/datum/antagonist/wizard/wizards
 	welcome_text = "You will find a list of available spells in your spell book. Choose your magic arsenal carefully.<br>In your pockets you will find a teleport scroll. Use it as needed."
 	flags = ANTAG_OVERRIDE_JOB | ANTAG_CLEAR_EQUIPMENT | ANTAG_CHOOSE_NAME | ANTAG_VOTABLE | ANTAG_SET_APPEARANCE
 	antaghud_indicator = "hudwizard"
+	required_age = 5
```
Comment (Geevies) [inline code/game/antagonist/outsider/wizard.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/station/changeling.dm [hunk 6]
```diff
@@ -16,6 +16,7 @@
 		"Zeng-Hu Mobility Frame",
 		"Bishop Accessory Frame"
 	)
+	required_age = 5
```
Comment (Geevies) [inline code/game/antagonist/station/changeling.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/station/cultist.dm [hunk 7]
```diff
@@ -32,6 +32,7 @@ var/datum/antagonist/cultist/cult
 	initial_spawn_req = 4
 	initial_spawn_target = 6
 	antaghud_indicator = "hudcultist"
+	required_age = 3
```
Comment (Geevies) [inline code/game/antagonist/station/cultist.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/station/loyalist.dm [hunk 8]
```diff
@@ -28,6 +28,7 @@ var/datum/antagonist/loyalists/loyalists
 	faction_indicator = "loyal"
 	faction_invisible = 1
 	restricted_jobs = list("AI", "Cyborg")
+	required_age = 7 // Head Loy important
```
Comment (Geevies) [inline code/game/antagonist/station/loyalist.dm]:
```suggestion
	required_age = 31
```

File: code/game/antagonist/station/revolutionary.dm [hunk 9]
```diff
@@ -30,6 +30,7 @@ var/datum/antagonist/revolutionary/revs
 
 	restricted_jobs = list("AI", "Cyborg")
 	protected_jobs = list("Security Officer", "Security Cadet", "Warden", "Detective", "Forensic Technician", "Head of Personnel", "Chief Engineer", "Research Director", "Chief Medical Officer", "Captain", "Head of Security", "Internal Affairs Agent")
+	required_age = 7 // head rev important
```
Comment (Geevies) [inline code/game/antagonist/station/revolutionary.dm]:
```suggestion
	required_age = 31
```

File: code/game/antagonist/station/rogue_ai.dm [hunk 10]
```diff
@@ -15,6 +15,7 @@ var/datum/antagonist/rogue_ai/malf
 	initial_spawn_req = 1
 	initial_spawn_target = 1
 	antaghud_indicator = "hudmalai"
+	required_age = 14 // arguably most important antag, needs a looong playtime
```
Comment (Geevies) [inline code/game/antagonist/station/rogue_ai.dm]:
```suggestion
	required_age = 31
```

File: code/game/antagonist/station/traitor.dm [hunk 11]
```diff
@@ -6,6 +6,7 @@ var/datum/antagonist/traitor/traitors
 	restricted_jobs = list("Internal Affairs Agent", "Head of Security", "Captain", "AI")
 	protected_jobs = list("Security Officer", "Security Cadet", "Warden", "Detective", "Forensic Technician")
 	flags = ANTAG_SUSPICIOUS | ANTAG_RANDSPAWN | ANTAG_VOTABLE
+	required_age = 3
```
Comment (Geevies) [inline code/game/antagonist/station/traitor.dm]:
```suggestion
	required_age = 10
```

File: code/game/antagonist/station/vampire.dm [hunk 12]
```diff
@@ -19,6 +19,7 @@ var/datum/antagonist/vampire/vamp = null
 		"Zeng-Hu Mobility Frame",
 		"Bishop Accessory Frame"
 	)
+	required_age = 5
```
Comment (Geevies) [inline code/game/antagonist/station/vampire.dm]:
```suggestion
	required_age = 10
```

**Comment by Arrow768** [comment]
Please ensure it is possible to assign people that would normally be locked out via the TP. 
If it is currently not possible, add a popup window that informs the admin about the issue (jobbanned, age-restricted, ...) and give them the option to assign them regardless (default to no)

**Comment by skull132** [comment]
> 
> 
> Please ensure it is possible to assign people that would normally be locked out via the TP.
> If it is currently not possible, add a popup window that informs the admin about the issue (jobbanned, age-restricted, ...) and give them the option to assign them regardless (default to no)

Per Geeves' analysis, this should be the case.

**Pull request merged**